theano/gpuarray/opt.py

```diff
@@ -1716,6 +1830,70 @@ def local_abstractconv_gradweights_gemm(node):
     return [rval]
 
 
+@local_optimizer([AbstractConv2d_gradWeights])
+def local_abstractconv_gemm_gradweights_alt(node):
```

Comment:
About this, it should be easy to do but it might become a little harder to read since there will be more if else conditions and also subsamples, filter_dilations, filter_flips and different ops  need to be handled for both 2d and 3d. Just wanted to check if it is alright before I proceed